### PR TITLE
Switch to setup-beam github action

### PR DESCRIPTION
**Pull request by starbelly**
- changes ci.yml to use setup-beam for all tests, except 19 which is still an ongoing issue with setup-beam

### CODE REVIEW

**Comment by starbelly** [comment]
We will wait until monday to merge so we can set the version in our config to `@v1`

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,23 +1,44 @@
...
-    container:
-      image: erlang:${{matrix.otp_vsn}}
+  test:
+    name: Erlang/OTP ${{matrix.otp}} / rebar3 ${{matrix.rebar3}}
+    runs-on: ubuntu-latest
     strategy:
       matrix:
-        otp_vsn: [19.3, 20.3, 21.3, 22.3, 23.2]
-        os: [ubuntu-latest]
+        otp: ['20.3.8.9', '21.3.8.9', '22.3.4.9', '23.3.1']
```
Comment (paulo-ferraz-oliveira) [inline .github/workflows/ci.yml]: Why not use '20', instead of '20.3.8.9'? The action will pick the latest for you 😄

Comment (starbelly) [reply .github/workflows/ci.yml]: It probably should yeah, I copied what was done in verl and that was just testing a good bit for the most part, but yeah agree to use majors only.

**Comment by starbelly** [comment]
We should go ahead and merge this, we can simply do another PR to change to `@v1` when its done.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,23 +1,44 @@
...
-      image: erlang:${{matrix.otp_vsn}}
+  test:
+    name: Erlang/OTP ${{matrix.otp}} / rebar3 ${{matrix.rebar3}}
+    runs-on: ubuntu-latest
     strategy:
       matrix:
-        otp_vsn: [19.3, 20.3, 21.3, 22.3, 23.2]
-        os: [ubuntu-latest]
+        otp: ['20', '21', '22', '23']
+        rebar3: ['3.14.1', '3.14.2', '3.14.3']
```
Comment (paulo-ferraz-oliveira) [inline .github/workflows/ci.yml]: FWIW, **latest** is 3.14.4.

**Review by paulo-ferraz-oliveira — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (starbelly) [reply .github/workflows/ci.yml]: Yeah, need to add.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -14,7 +14,7 @@ jobs:
     strategy:
       matrix:
         otp: ['20', '21', '22', '23']
-        rebar3: ['3.14.1', '3.14.2', '3.14.3']
+        rebar3: ['3.14.1', '3.14.2', '3.14.3', '3.14.3']
```
Comment (paulo-ferraz-oliveira) [inline .github/workflows/ci.yml]:
```suggestion
        rebar3: ['3.14.1', '3.14.2', '3.14.3', '3.14.4']
```

Comment (paulo-ferraz-oliveira) [reply .github/workflows/ci.yml]: 😄

Comment (starbelly) [reply .github/workflows/ci.yml]: One of those days 😁

Comment (paulo-ferraz-oliveira) [reply .github/workflows/ci.yml]: I kinda guessed...

**Review by ericmj — APPROVED** [review]

**Pull request merged**